carball/analysis/stats/controls/controls.py

```diff
@@ -0,0 +1,37 @@
+from typing import Dict
+
+import pandas as pd
+
+from ....analysis.stats.stats import BaseStat
+from ....generated.api import game_pb2
+from ....generated.api.player_pb2 import Player
+from ....generated.api.stats.player_stats_pb2 import PlayerStats
+from ....json_parser.game import Game
+
+
+class ControlsStat(BaseStat):
+
+    def calculate_player_stat(self, player_stat_map: Dict[str, PlayerStats], game: Game, proto_game: game_pb2.Game,
+                              player_map: Dict[str, Player], data_frame: pd.DataFrame):
+        for player_key, stats in player_map.items():
+            player_name = player_map[player_key].name
+            steer_values = data_frame[player_name]['steer'].tolist()
```

Comment:
This ties into what I said before, but in general try to keep the data in pandas and use aggregate functions. `tolist()` is very intensive.